keps/sig-auth/2799-token-controller-deprecation/README.md

```diff
@@ -0,0 +1,395 @@
+  - [Monitoring Requirements](#monitoring-requirements)
+  - [Dependencies](#dependencies)
+  - [Scalability](#scalability)
+  - [Troubleshooting](#troubleshooting)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (Optional)](#infrastructure-needed-optional)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+Items marked with (R) are required _prior to targeting to a milestone /
+release_.
+
+- [x] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [x] (R) KEP approvers have approved the KEP status as `implementable`
+- [x] (R) Design details are appropriately documented
+- [x] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+- [x] (R) Graduation criteria is in place
+- [x] (R) Production readiness review completed
+- [x] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This KEP proposes a namespace-scoped and service-account-scoped binary label to
+allow users to opt in/out the provision of secret-based service account tokens
+in Token Controller. In addition, it sketches out the phases to deprecate Token
+Controller.
+
+## Motivation
+
+As BoundServiceAccountTokenVolume is GA in 1.22, pods’ service account tokens
+would be obtained via TokenRequest API and stored as projected volume. This
+change obviates the need for auto-generation of secret-based service account
+tokens in Token Controller. The secret-based tokens are [not secure by design](https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/1205-bound-service-account-tokens#background)
+and the token controller is [fragile by design in some cases](https://github.com/kubernetes/kubernetes/issues/98474)
+where it is unable to handle the churns between secrets and service account
+controller loops.
+
+### Goals
+
+- No auto-generation of secret-based service account token.
+- Removal of secret-based service account token that are referenced in service
+  accounts.
+
+### Non-Goals
+
+## Proposal
+
+- In release 1.23, change the service account control loop in Token Controller
+  to not create secret for the service account. At the same time, warn usage of
+  auto-created secret-based service account tokens and enrouage users to use
+  TokenRequest API or manually-created secret-based service account tokens.
```

Comment:
why " that are not currently mounted into pods"?

even if in 1.24+, users can still mount secrets that are auto-created in 1.22- (or auto-created like secrets crafted by users, i.e. any bi-directionally referenced secrets with service accounts) into pods manually. for this case, we need to notify users to use manually created tokens instead so that we can clean up.